core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2389,32 +2401,31 @@ class KafkaApis(val requestChannel: RequestChannel,
       true
   }
 
-  def handleElectPreferredReplicaLeader(request: RequestChannel.Request): Unit = {
+  def handleElectReplicaLeader(request: RequestChannel.Request): Unit = {
+
+    val electionRequest = request.body[ElectLeadersRequest]
+    val partitions: Set[TopicPartition] = if (electionRequest.data().topicPartitions() == null) {
```

Comment:
I wonder if we ought to move this after the auth check so that we're less likely to inadvertently leak topic names.

Reply:
Good catch. I also fixed the authorization branch so that we don't generate a map of partitions to error when the request version is after version 0.